### PR TITLE
Slime Core Refinery, but for real this time 

### PR DESCRIPTION
**Pull request by MajManatee**
# Document the changes in your pull request

Adds a brand new machine to xenobiology, the Plort redemption machine. It convert slime cores into research points, allowing xenobio to do something while they work on their powergaming.

# Spriting

auauauauauauau

Thanks to @cuackles for the sprite! 
# Wiki Documentation

Yeah there probably needs to be a wiki thing made for a few things
Point values for each slime core need to be added
Machine needs to be added

# Changelog

:cl:  @cuackles @MajManatee 
rscadd: Adds the Plort Redemption machine
wip: This is work in progress  
mapping: puts it in xenobio
/:cl:

### CODE REVIEW

**Comment by Cartlord** [comment]
What reason does the Xenobiologist have to not just make a horrifying number of grey cores?

**Comment by MajManatee** [comment]
to spend an hour and a half to become an unkillable god
but now they have a choice to be useful!

File: code/modules/research/machinery/plortmachine.dm [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+/obj/machinery/plortrefinery
+	name = "plort redemption machine"
+	desc = "A machine that accepts slime cores, and sells them to the highest bidder. This generates research, depending on the rarity."
+	icon = 'icons/obj/machines/mining_machines.dmi' // TODO	
+	icon_state = "ore_redemption" // TODO
+	layer = BELOW_OBJ_LAYER
+	density = TRUE
+	speed_process = TRUE
+	var/research_upgrade = 1
```
Comment (ToasterBiome) [inline code/modules/research/machinery/plortmachine.dm]: bad variable name, doesn't describe what it does

File: code/modules/research/machinery/plortmachine.dm [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+/obj/machinery/plortrefinery/accept_check(obj/item/O)
+	if(istype(O, /obj/item/slime_extract))
+		return TRUE
+	else
+		return FALSE
+
+/obj/machinery/plortrefinery/examine(mob/user)
+	. = ..()
+	if(in_range(user, src) || isobserver(user))
+		. += "<span class='notice'>The status display reads: Converting cores at <b>[research_upgrade*100]%</b> their value.<span>"
```
Comment (ToasterBiome) [inline code/modules/research/machinery/plortmachine.dm]: Doesn't convert, says it sells them

**Review by ToasterBiome — CHANGES_REQUESTED** [review]

File: code/modules/research/machinery/plortmachine.dm (hunk 2 above)
Comment (MajManatee) [reply code/modules/research/machinery/plortmachine.dm]: Will be fixed

File: code/modules/research/machinery/plortmachine.dm (hunk 1 above)
Comment (MajManatee) [reply code/modules/research/machinery/plortmachine.dm]: youre a bad variable, but ok
will be fixed

**Comment by MajManatee** [comment]
Just waiting for sprites. Works as of now

**Comment by Simplehorror** [comment]
why this draft

**Comment by MajManatee** [comment]
because it wasnt done
now it is

**Comment by drderp3635** [comment]
Still should be called an extract redemption machine. Reds should also be 50 to match other 2nd tier slimes.

**Comment by MajManatee** [comment]
red value changed, oops
as for the name. no.

**Review by ToasterBiome — CHANGES_REQUESTED** [review]
other maps, at least the ones in rotation

**Comment by MajManatee** [comment]
oh yeah, forgor

**Comment by MajManatee** [comment]
still waiting on sprite from my local spriter

**Comment by Firewolf34** [comment]
Maybe a future improvement is to have goals for certain cores, which provide a bonus?

Really think Science department needs actionable goals other than powergaming

**Comment by MajManatee** [comment]
What do you mean goals?

**Review by ToasterBiome — APPROVED** [review]
map fine

**Review by JamieD1 — APPROVED** [review]

**Pull request merged**